### PR TITLE
Fixes race condition on ELB checking instance availability (reported in #383)

### PR DESCRIPTION
**Pull request by jsmartin**
This patch uses the ELB as the source of truth (when applicable) for the health state of an instance in an ASG.  This negates any race condition.

This will also wait for a new ASG to have it's instances be healthy, as deemed by the ELB.

@bwhaley please review.

### CODE REVIEW

**Comment by bwhaley** [comment]
Would like to review but am unable until 1/17 or so due to vacation.

**Comment by jsmartin** [comment]
no problem. enjoy the vacay.

**Comment by bwhaley** [comment]
Per discussion via IRC, @jsmartin will add a call to `wait_for_elb()` for new ASGs in addition to when replacing instances. The logic looks good otherwise.

**Comment by jsmartin** [comment]
@bwhaley Added some new commits and docs.  wait_for_elb now optionally happens on new/updated ASGs as well.

**Comment by bwhaley** [comment]
I haven't had a chance to test but code LGTM!

**Comment by rwky** [comment]
This doesn't work with replacing instances, I've submitted a pull request to @jsmartin repo jsmartin/ansible-modules-core/pull/1

File: cloud/amazon/ec2_asg.py [hunk 1]
```diff
@@ -403,6 +461,8 @@ def delete_autoscaling_group(connection, module):
             time.sleep(10)
 
         group.delete()
+        while len(connection.get_all_groups(names=[group_name])):
+                time.sleep(5)
```
Comment (abadger) [inline cloud/amazon/ec2_asg.py:465]: Extra indent here.

**Comment by abadger** [comment]
Okay, I rebased this onto the tip of devel and then merged this in.  @rwky, I also pulled in your changes but I squashed them into a single commit.  The rebasing and squashing were because I also pulled this into the stable-1.9 branch and it was easier to cherry-pick the commits that way.  thanks guys for creating the code and for testing and reviewing each others' changes.

# Merged

Hi!

This has been merged in, and will also be included in the next major release.

If you or anyone else has any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!

**Pull request closed without merging**